pimcore/lib/Pimcore/Bundle/AdminBundle/Controller/Reports/CustomReportController.php

```diff
@@ -187,7 +187,7 @@ public function columnConfigAction(Request $request)
         $result = [];
 
         try {
-            $adapter = CustomReport\Config::getAdapter($configuration);
+            $adapter = $this->getReportAdapter($configuration->type, $configuration);
```

Comment:
I think we should stick with `CustomReport\Config::getAdapter()` here to make sure the fallback logic for an empty type still works and remove the `getReportAdapter()` method from this controller. We can move the logic in the static method to a custom service at some point but for now i'd just stick to the static method.

Reply:
I dont think thats an good idea, the static method uses directly the container to build the adapter, which is a big no-go for me. I just kept it for BC and marked it deprecated.

we could introduce a new factory to create factories :D, but thats kinda weird isn't it? I mean, basically the controller should be responsible to take care about the fallback, shouldn't it? if somebody else uses the service-locator, he should be responsible for fallbacks then. also: configurations should always be complete, otherwise you are producing inconsistent data.

Reply:
The thing is we can't assume that configurations are complete. I completely agree with your points (basically every `\Pimcore::getContainer()` call is "wrong") and I think that that the loading and fallback should be encapsulated somewhere, but that's something which can be done later. The adapter loading should be consistent - I'd rather stick to the static method for now than duplicating the code in the controller.

For now let's just stick to the static method to have the same logic to load adapters throughout the system and improve that at a later point.

Reply:
Alright, at your risk ;) but you owe me a beer for that 🍺  😉 

Will do that later.

Reply:
Deal ;)